### PR TITLE
test_models: support for specifying segment

### PR DESCRIPTION
**Pull request by sshane**
Pre-req for PR: https://github.com/commaai/openpilot/pull/24173

### CODE REVIEW

File: selfdrive/car/tests/routes.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
   TestRoute("8190c7275a24557b|2020-01-29--08-33-58", CHRYSLER.PACIFICA_2019_HYBRID),
   TestRoute("3d84727705fecd04|2021-05-25--08-38-56", CHRYSLER.PACIFICA_2020),
 
-  #TestRoute("f1b4c567731f4a1b|2018-04-30--10-15-35", FORD.FUSION),
+  # TestRoute("f1b4c567731f4a1b|2018-04-30--10-15-35", FORD.FUSION),
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/routes.py:39]: why was this changed? the convention tends to be no spaces for commented out code

Comment (sshane) [reply selfdrive/car/tests/routes.py:39]: The convention is to always include 1 space after a comment, according to PEP8. Black does this as well

Comment (adeebshihadeh) [reply selfdrive/car/tests/routes.py:39]: we don't adhere to PEP8

File: selfdrive/car/tests/test_models.py [hunk 2]
```diff
@@ -36,32 +36,37 @@
...
         print(f"Skipping tests for {cls.car_model}: missing route")
         raise unittest.SkipTest
       raise Exception(f"missing test route for {cls.car_model}")
 
     disable_radar = False
-    for seg in (2, 1, 0):
+
+    test_segs = [2, 1, 0]
+    if cls.test_route.segment is not None:
+      test_segs.insert(0, cls.test_route.segment)
```
Comment (adeebshihadeh) [inline selfdrive/car/tests/test_models.py:65]: we only want to run on the specified segment if one is specified
